spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/MappingSpannerWriteConverter.java

```diff
@@ -156,19 +160,36 @@ private void writeProperty(WriteBuilder sink, PersistentPropertyAccessor accesso
 		Class<?> propertyType = property.getType();
 		ValueBinder<WriteBuilder> valueBinder = sink.set(property.getColumnName());
 
-		boolean valueSet;
+		boolean valueSet = false;
 
 		/*
-		 * Due to type erasure, binder methods for Iterable properties must be manually specified.
-		 * ByteArray must be excluded since it implements Iterable, but is also explicitly
-		 * supported by spanner.
+		 * Due to type erasure, binder methods for Iterable properties must be manually
+		 * specified. ByteArray must be excluded since it implements Iterable, but is also
+		 * explicitly supported by spanner.
+		 *
+		 * Also due to type erasure, custom conversions cannot be used for iterables
+		 * because different conversions between lists of different types cannot be
+		 * distinguished.
 		 */
 		if (ConversionUtils.isIterableNonByteArrayType(propertyType)) {
 			valueSet = attemptSetIterableValue((Iterable) propertyValue, valueBinder,
 					property);
 		}
 		else {
-			valueSet = attemptSetSingleItemValue(propertyValue, valueBinder, property);
+			valueSet = attemptSetSingleItemValue(propertyValue, valueBinder,
+					propertyType);
+			if (!valueSet) {
+				for (Class targetType : this.singleItemType2ToMethodMap.keySet()) {
+					if (!canConvert(propertyType, targetType)) {
```

Comment:
nit: same as above, I'd just move it inside `attemptSetSingleItemValue`

Reply:
I will include this change in my next PR! (I didn't wanna make this LGTM stale by doing it here haha)